broadcaster/backlog/backlog.go

```diff
@@ -75,6 +97,10 @@ func (b *backlog) Append(bm *m.BroadcastMessage) error {
 
 		prevMsgIdx := segment.End()
 		if segment.count() >= b.config().SegmentLimit {
+			segment.messagesLock.RLock()
+			msg.CumulativeSumMsgSize = segment.messages[len(segment.messages)-1].CumulativeSumMsgSize
```

Comment:
Shouldn't this be `+=` not `=`? Let's also skip this if `len(segment.messages) == 0` just in case

Reply:
basically we want to carry over the previous segment's last message's cumulative sum to the newly created segment's first message for correctness, so we initialize it first and then when `err := segment.append(prevMsgIdx, msg)` is called the correct cumulative sum is calculated.

Reply:
Ah I see, yeah I misunderstood how UpdateCumulativeSumMsgSize worked